<launch>
  <machine name="rock64" address="rock64" user="rock64" password="rock64" env-loader="/home/rock64/catkin_ws/devel/env.sh" default="true"/>

	<node name="dynamixel_manager" pkg="servo_controller" type="controller_manager.py" required="true" output="screen" machine="rock64">
        <rosparam>
            namespace: dxl_manager
            serial_ports:
                steer_port:
                    port_name: "/dev/ttyUSB0"
                    baud_rate: 115200
                    min_motor_id: 0
                    max_motor_id: 1
                    update_rate: 20
        </rosparam>
    </node>

	<rosparam file="$(find servo_controller)/config/dynamixel_joint_controllers.yaml" command="load"/>
    
    <!-- start specified joint controllers -->
<!--    <node name="dynamixel_controller_spawner" pkg="dynamixel_controllers" type="controller_spawner.py" -->
    <node name="dynamixel_controller_spawner" pkg="servo_controller" type="controller_spawner.py" machine="rock64"
          args="--manager=dxl_manager
                --port=steer_port
                --type=simple
                front_controller
                back_controller"
          output="screen"/>

<node pkg="vesc_driver" type="vesc_driver_node" name="vesc_node" respawn="true"
        output="screen" machine="rock64" >
    <param name="port" value="/dev/vesc" />
  </node>

	<node pkg="drive_control" type="control_node.py" name="control_node" machine="rock64"/>

  
  <arg name="joy_config" default="xbox" />
  <arg name="joy_dev" default="/dev/input/js0" />
  <arg name="config_filepath" default="$(find drive_control)/config/$(arg joy_config).config.yaml" />
  <arg name="joy_topic" default="joy" />
  
  <node pkg="joy" type="joy_node" name="joy_node" machine="rock64">
    <param name="dev" value="$(arg joy_dev)" />
    <param name="deadzone" value="0.3" />
    <param name="autorepeat_rate" value="20" />
    <remap from="joy" to="$(arg joy_topic)" />
  </node> 

  <node pkg="drive_control" name="teleop_drive_joy" type="teleop_drive_joy_node" machine="rock64">
    <rosparam command="load" file="$(arg config_filepath)" />
    <remap from="joy" to="$(arg joy_topic)" />
  </node>

</launch>
